### PR TITLE
Drop replace methods with builtin

### PR DESCRIPTION
**Pull request by ericvergnaud**
## Changes
Our code implements 'replace' methods on some dataclasses with no benefit over the builtin method.
This PR fixes that.

### Linked issues
None

### Functionality
None

### Tests
- [x] ran unit tests

### CODE REVIEW

File: src/databricks/labs/ucx/source_code/base.py [hunk 1]
```diff
@@ -41,25 +42,6 @@ class Advice:
     end_line: int
     end_col: int
 
-    def replace(
```
Comment (nfx) [inline src/databricks/labs/ucx/source_code/base.py:44]: what about directfs accesses?

**Review by nfx — CHANGES_REQUESTED** [review]

File: src/databricks/labs/ucx/source_code/base.py (hunk 1 above)
Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/base.py:44]: already done in https://github.com/databrickslabs/ucx/pull/2599 and https://github.com/databrickslabs/ucx/pull/2630

Comment (ericvergnaud) [reply src/databricks/labs/ucx/source_code/base.py:44]: what is left re directfs is more focused methods that call dataclasses.replace

**Review by nfx — APPROVED** [review]
lgtm

**Pull request merged**